src/Domain/Aggregates/User.cs

```diff
@@ -11,4 +12,7 @@ public sealed class User(UserId id) : AggregateRoot<UserId>(id)
     public string Email { get; init; } = default!;
 
     public string PasswordHash { get; init; } = default!;
+    
+
+    public Wallet Wallet { get; init; } = default!;
```

Comment:
while default assignment is good in 80% of the cases, we could assign a default wallet with 0 balance to the user. so that it is not required to manually initialize an empty wallet every time we create a user.